packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -5364,21 +5981,64 @@ export class Matrix {
 
     /**
      * Returns the matrix as a Float32Array or Array<number>
-     * Example Playground - https://playground.babylonjs.com/#AV9X17#49
-     * @returns the matrix underlying array
+     * @deprecated Use asArray
      */
-    public toArray(): DeepImmutable<Float32Array | Array<number>> {
-        return this._m;
+    public toArray(): FloatArray;
+
+    /**
+     * Stores the matrix in a Float32Array or Array<number>
+     * Example Playground - https://playground.babylonjs.com/#AV9X17#49
+     * @param array The destination array
+     * @param index The destination index to start ay
+     * @returns the matrix
+     */
+    public toArray(array: FloatArray, index: number): this;
+    public toArray(array: Nullable<FloatArray> = null, index: number = 0): this | FloatArray {
+        if (!array) {
+            return this._m;
+        }
+        const m = this._m;
+        for (let i = 0; i < m.length; i++) {
+            array[index + i] = m[i];
+        }
+        return this;
     }
+
     /**
      * Returns the matrix as a Float32Array or Array<number>
      * Example Playground - https://playground.babylonjs.com/#AV9X17#114
      * @returns the matrix underlying array.
      */
-    public asArray(): DeepImmutable<Float32Array | Array<number>> {
+    public asArray(): Tuple<number, 16> {
         return this._m;
     }
 
+    public fromArray(array: FloatArray, index: number = 0): this {

```

Comment:
Doc is missing for the new methods.

Reply:
Same doc comment as above, but for `Matrix`.